tests/unit/dataactvalidator/test_c9_award_financial.py

```diff
@@ -1,37 +1,151 @@
-from tests.unit.dataactcore.factories.staging import AwardFinancialFactory, AwardFinancialAssistanceFactory
+from tests.unit.dataactcore.factories.staging import AwardFinancialFactory
+from tests.unit.dataactcore.factories.staging import AwardFinancialAssistanceFactory
 from tests.unit.dataactvalidator.utils import number_of_errors, query_columns
 
+
 _FILE = 'c9_award_financial'
+_TAS = 'c9_award_financial_tas'
+
 
 def test_column_headers(database):
     expected_subset = {'row_number', "fain", "uri"}
     actual = set(query_columns(_FILE, database))
     assert (actual & expected_subset) == expected_subset
 
 
-def test_success(database):
-    """ Test case where all fain and uri in D2 are present in C """
-
-    # Test with only one of fain/uri
-    award_fin_fain = AwardFinancialFactory(uri = None)
-    award_fin_uri = AwardFinancialFactory(fain = None)
-    afa_fain = AwardFinancialAssistanceFactory(fain = award_fin_fain.fain, uri = None, federal_action_obligation = 123)
-    afa_uri = AwardFinancialAssistanceFactory(fain = None, uri = award_fin_uri.uri, federal_action_obligation = 234)
-    # Test with both fain and uri
-    award_fin = AwardFinancialFactory()
-    afa = AwardFinancialAssistanceFactory(fain = award_fin.fain, uri = award_fin.uri, federal_action_obligation = 345)
-    # Test with zero federal action obligation
-    afa_zero = AwardFinancialAssistanceFactory(federal_action_obligation = 0)
-
-    assert number_of_errors(_FILE, database, models=[award_fin_fain, award_fin_uri, afa_fain, afa_uri, afa_zero,
-                                                     award_fin, afa]) == 0
-
-def test_failure(database):
-    """ Test fain and uri present in D2 but not in C """
-    award_fin = AwardFinancialFactory(fain = None, uri = None)
-    afa_fain = AwardFinancialAssistanceFactory(uri = None, federal_action_obligation = 123)
-    afa_uri = AwardFinancialAssistanceFactory(fain = None, federal_action_obligation = 234)
-    # Test with both fain and uri
-    afa = AwardFinancialAssistanceFactory(federal_action_obligation = 345)
-
-    assert number_of_errors(_FILE, database, models=[award_fin, afa_fain, afa_uri, afa]) == 3
+def test_equal_fain(database):
+    """Tests that File D2 (award financial assistance) fain matches
+    File C (award financial) fain."""
+    tas = _TAS
+    afa = AwardFinancialAssistanceFactory(
+        tas=tas, fain='abc', uri=None, federal_action_obligation='1', original_loan_subsidy_cost='1')
+    af = AwardFinancialFactory(
+        tas=tas, submisson_id=afa.submission_id, fain=afa.fain, uri=None)
+
+    errors = number_of_errors(_FILE, database, models=[afa, af])
+    assert errors == 0
+
+
+def test_equal_uri(database):
+    """Tests that File D2 (award financial assistance) uri matches
+    File C (award financial) uri."""
+    tas = _TAS
+    afa = AwardFinancialAssistanceFactory(
+        tas=tas, fain=None, uri='xyz', federal_action_obligation='1', original_loan_subsidy_cost='1')
```

Comment:
Might be good to have one of these tests set e.g. federal_action_obligation to 0 to make sure the logic is working correctly that only one needs to be nonzero